### ISSUE TITLE
Dynamic importing dynamic versions

### ISSUE THREAD

**Issue by arggh** [meteor/meteor-feature-requests#351]
(I couldn't resist writing that title.)

**TL;DR** Using dynamic imports can quickly increase the size of `dynamic-versions.js`, which is loaded in the static bundle. Could we somehow make it smaller or loaded dynamically?

-----

I've been refactoring a rather large app to benefit from dynamic imports. I started with about 5Mb payload of JS without dynamic imports. Cutting away most of the app except login, signup and few necessities got me to around 985kb. From there on, I started to include routes and components via dynamic imports.

Soon I noticed that the `dynamic-versions.js` file, which gets shipped with the static bundle, was slowly but surely growing and growing. It's now almost 200kb.

When aiming for < 500kb initial payload, especially with a Meteor app, that 200kb is a lot!

I wonder if there is a way to reduce the size of that file?

**Comment by mariusrak** [comment]
I have no `dynamic-versions.js` file in my bundle. However I have a file `app.js` which includes all files that are available in the application even those that were not requested yes. You mean this file?

If so, I agree with You, that it would be great if it was not needed to send whole tree to user but rather load it as further parts are fetched.

**Comment by mitar** [comment]
Is 200kb even when compressed?
